src/plugins/usage_collection/server/collector/collector.ts

```diff
@@ -46,26 +47,71 @@ export type MakeSchemaFrom<Base> = {
    */
   esClient: ElasticsearchClient;
   /**
-   * Request-scoped Saved Objects client:
-   * - When users are requesting a sample of data, it is scoped to their role to avoid exposing data they should't read
-   * - When building the telemetry data payload to report to the remote cluster, the requests are scoped to the `kibana` internal user
+   * Request-scoped Saved Objects client
+   * @remark Bear in mind when testing your collector that your user has the same privileges as the Kibana Internal user to ensure the expected data is sent to the remote cluster (more info: {@link CollectorFetchContext})
    */
   soClient: SavedObjectsClientContract | ISavedObjectsRepository;
+} & (WithKibanaRequest extends true
+  ? {
+      /**
+       * The KibanaRequest that can be used to scope the requests:
+       * It is provided only when your custom clients need to be scoped. If not available, you should use the Internal Client.
+       * More information about when scoping is needed: {@link CollectorFetchContext}
+       * @remark You should only use this if you implement your collector to deal with both scenarios: when provided and, especially, when not because it's the actual situation when the telemetry payload is sent to the remote service.
+       */
+      kibanaRequest?: KibanaRequest;
+    }
+  : {});
+
+export type CollectorFetchMethod<
+  WithKibanaRequest extends boolean | undefined,
+  TReturn,
+  ExtraOptions extends object = {}
+> = (
+  this: Collector<TReturn, unknown> & ExtraOptions, // Specify the context of `this` for this.log and others to become available
+  context: CollectorFetchContext<WithKibanaRequest>
+) => Promise<TReturn> | TReturn;
+
+export interface ICollectorOptionsFetchExtendedContext<WithKibanaRequest extends boolean> {
+  /**
+   * Set to `true` if your `fetch` method requires the `KibanaRequest` object to be added in its context {@link CollectorFetchContextWithRequest}.
+   * @remark You should fully understand acknowledge that by using the `KibanaRequest` in your collector, you need to ensure it should specially work without it because it won't be provided when building the telemetry payload actually sent to the remote telemetry service.
+   */
+  kibanaRequest?: WithKibanaRequest;
 }
 
-export interface CollectorOptions<T = unknown, U = T> {
+export type CollectorOptionsFetchExtendedContext<
+  WithKibanaRequest extends boolean
+> = ICollectorOptionsFetchExtendedContext<WithKibanaRequest> &
+  (WithKibanaRequest extends true // If enforced to true via Types, the config must be expected
+    ? Required<Pick<ICollectorOptionsFetchExtendedContext<WithKibanaRequest>, 'kibanaRequest'>>
+    : {});
+
+export type CollectorOptions<
+  TFetchReturn = unknown,
+  UFormatBulkUploadPayload = TFetchReturn, // TODO: Once we remove bulk_uploader's dependency on usageCollection, we'll be able to remove this type
```

Comment:
Recommendation: We have a growing list of `TODO`'s that are waiting on removing the `bulk_uploader's` dependency on `usageCollection`. Can we add a meta issue with all these `TODO`'s and link the meta issue to the one for removing the `bulk_uploader`?

Reply:
Good one! I've created https://github.com/elastic/kibana/issues/83521 to track them :)